### PR TITLE
feat(core): trigger file to initiate snapshot recovery

### PR DESCRIPTION
**Pull request by jerrinot**
Previously, snapshot recovery was driven by the Cairo Snapshot Instance ID property, which had no default value and was generally confusing to use. Snapshot recovery is now triggered by the existence of a trigger file in the QuestDB root directory: If a file named `root/_restore` exists and a snapshot is available, it will be restored on startup.

The old strategy with Cairo Snapshot Instance IDs is left of backwards compatibility, but it's not recommended. 

Additionally, a failure during snapshot recovery aborts QuestDB start. This is technically a breaking change - since before it would just log a failure and continue.  However, a failure during recovery means the system state is undefined and should not be relied on.

TODO:
- [x] fail-fast when a trigger file exists, but there is no snapshot
- [x] warning when cairo id is set - it has no effect and have to communicate this to a user

Docs PR: https://github.com/questdb/documentation/pull/26

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 6 / 8 (75.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/DatabaseSnapshotAgentImpl.java|6|8|75.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 39 / 46 (84.78%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/DatabaseSnapshotAgentImpl.java|8|14|57.14%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|29|30|96.67%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|1|1|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 39 / 46 (84.78%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/DatabaseSnapshotAgentImpl.java|8|14|57.14%|
|:large_blue_circle:|io/questdb/PropServerConfiguration.java|29|30|96.67%|
|:large_blue_circle:|io/questdb/cairo/DefaultCairoConfiguration.java|1|1|100.00%|
|:large_blue_circle:|io/questdb/cairo/CairoConfigurationWrapper.java|1|1|100.00%|

**Comment by sklarsa** [comment]
@jerrinot @bluestreak01 we need to modify our k8s helm chart, cloud, and docs to account for this change. For users to restore a database, they need to create an init container that runs before QuestDB is started and creates this marker file in the database directory

**Comment by jerrinot** [comment]
@sklarsa I just pushed an update. it's adding a new trigger - a file. but the old way - via cairo snapshot id - works too.

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 7 / 12 (58.33%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/DatabaseSnapshotAgentImpl.java|7|12|58.33%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 23 / 42 (54.76%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/DatabaseSnapshotAgentImpl.java|23|42|54.76%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 32 / 42 (76.19%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/DatabaseSnapshotAgentImpl.java|32|42|76.19%|

File: core/src/main/java/io/questdb/cairo/DatabaseSnapshotAgentImpl.java [hunk 1]
```diff
@@ -511,8 +511,12 @@ void recoverSnapshot() {
             }
 
             // OK, we need to recover from the snapshot.
+            if (triggerExists) {
+                LOG.info().$("starting snapshot recovery [trigger=file]").$();
+            }
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/cairo/DatabaseSnapshotAgentImpl.java:516]: need else block!

**Review by bluestreak01 — APPROVED** [review]

**Review by bluestreak01 — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 41 / 51 (80.39%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/DatabaseSnapshotAgentImpl.java|41|51|80.39%|

**Pull request merged**